src/qibo/ui/mpldrawer.py

```diff
@@ -0,0 +1,654 @@
+                temp_mgates = []
+
+            cluster_gates.append([item])
+
+    if len(temp_gates) > 0:
+        cluster_gates.append(temp_gates)
+
+    if len(temp_mgates) > 0:
+        cluster_gates.append(temp_mgates)
+
+    return cluster_gates
+
+
+def _check_list_str(substrings, string):
+    return any(item in string for item in substrings)
+
+
+def _process_gates(array_gates):
+
+    if len(array_gates) == 0:
+        return []
+
+    gates_plot = []
+
+    for gate in array_gates:
+        init_label = gate.name.upper()
+
+        if init_label == "CCX":
+            init_label = "TOFFOLI"
+        elif init_label == "CX":
+            init_label = "CNOT"
+        elif _check_list_str(["SX", "CSX"], init_label):
+            is_dagger = init_label[-2:] == "DG"
+            init_label = (
+                r"$\rm{\sqrt{X}}^{\dagger}$" if is_dagger else r"$\rm{\sqrt{X}}$"
+            )
+        elif (
+            len(gate._control_qubits) > 0
+            and "C" in init_label[0]
+            and "CNOT" != init_label
+        ):
+            init_label = gate.draw_label.upper()
+
+        if init_label in [
+            "ID",
+            "MEASURE",
+            "KRAUSCHANNEL",
+            "UNITARYCHANNEL",
+            "DEPOLARIZINGCHANNEL",
+            "READOUTERRORCHANNEL",
+        ]:
+            for qbit in gate._target_qubits:
+                item = (init_label,)
+                item += ("q_" + str(qbit),)
+                gates_plot.append(item)
+        elif init_label == "ENTANGLEMENTENTROPY":
+            for qbit in list(range(circuit.nqubits)):
+                item = (init_label,)
+                item += ("q_" + str(qbit),)
+                gates_plot.append(item)
+        else:
+            item = ()
+            item += (init_label,)
+
+            for qbit in gate._target_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            for qbit in gate._control_qubits:
+                if qbit is tuple:
+                    item += ("q_" + str(qbit[0]),)
+                else:
+                    item += ("q_" + str(qbit),)
+
+            gates_plot.append(item)
+
+    return gates_plot
+
+
+def _plot_params(style: Union[dict, str, None]) -> dict:
+    if not isinstance(style, dict):
+        try:
+            style = STYLE.get(style) if (style is not None) else STYLE["default"]
+        except AttributeError:
+            style = STYLE["default"]
+
+    return style
+
+
+def plot(circuit, scale=0.6, cluster_gates=True, style=None):
+    """Main matplotlib plot function for Qibo circuit
+    circuit         A Qibo circuit to plot (type: qibo.models.circuit.Circuit)
+    scale           Scale the ouput plot
+    cluster_gates   Group single-qubit gates
+    style           Style applied to the circuit (built-in styles: garnacha, fardelejo, quantumspain, color-blind and cachirulo)
```

Comment:
I don't know how good it will be, but it would be very nice to have a style based on  [SciencePlots](https://github.com/garrettj403/SciencePlots), to produce circuit diagrams standardized with the default LaTeX font. Also for this, we can keep it for future PR.

Reply:
This is a good enhance but I think it should be better to keep it for a future PR